### DISCUSSION TITLE
Suitability of fast event related design with variable time of a decision event

### DISCUSSION THREAD

**Discussion by gtertikas** [cvnlab/GLMsingle#135] [General]
Hello,

I am relatively new to using the GLMsingle toolbox and have a couple of questions regarding the formatting of my data and the design matrix.

**Variable Presentation Time of Stimuli:**

I have a dataset where the presentation time of stimuli varies based on decision-making processes (i.e. stimuli changes when the participant has pressed a button). These events are not the primary focus of our analysis. Could you provide guidance on how to format my data to accommodate this variability? Are there specific steps or considerations within GLMsingle to handle variable presentation times effectively in this case? My TR is 1.5 sec for reference (multiband sequence of factor 3) and these events can range from 0.2 sec to 8 or more seconds. 

**Handling of Concurrent Events:**
In my experiment, there are instances where two events occur simultaneously. Is there a way to control for this in the design matrix? Any recommendations on how to properly account for or manage overlapping events within GLMsingle would be greatly appreciated.

Thank you for your assistance!

Best regards,
George

**Comment by kendrickkay** [comment]
Regarding variable presentation time, the main issue to consider is your belief about what the associated BOLD response might be to these variable duration events.  A standard presumption could be that a longer event (e.g. 8 or more) would correspond to a "convolved/longer-duration" timecourse.   An issue that is covered on the GLMsingle FAQ is how one might accommodate different duration events in GLMsingle.  The short answer is that it's a little challenging, since GLMsingle wants to think in terms of discrete events (that are all comparable in terms of their associated BOLD timecourse).

However, since you indicated that you aren't fundamentally interested in the BOLD responses associated with these variable events, things are less pressing.  Perhaps you can just code the events as one or more short events and plan to just discard the estimated single-trial amplitude associated with these events.  Also, bear in mind that over some range of durations the timecourse differences between, e.g., 1 s and 2 s neural events, correspond to fairly small hemodynamic timecourse shape differences.  Hence, there is some amount of approximation that one can do with probably minimal inaccuracy.  So, for example, you might code long events (e.g. a 8-s event) as a series / train of 1-s events, and code a very short event (e.g. 0.5 s) as just a 1-s event.

Regarding concurrent events, the architecture of GLMsingle really wants you to treat the case of simultaneous events as itself a wholly novel type of condition.  (It is not easy to do the opposite, namely, act as if the response to a trial with simultaneous events is (e.g.) the sum of the responses to the individual events (assuming that is something you were envisioning).)  

Hopefully this all makes sense.

**Reply by gtertikas** [reply]
Thank you so much for your fast reply! I can understand that there will be rounding of events to the closest 1 second so as to account for varying durations (after splitting longer duration to smaller events).

I just have some follow up questions in regards to that:

1) In terms of the duration of the neural events, does it then make sense to make the short events in the order of 1 second because making them shorter would not make the function of the BOLD response reliable anymore? How would I model some events that I have with duration of 7.3 seconds for example?
2) Does the TR matter in terms of the matrix of events (i.e. 1.5 TR means 1 second events need to be split in half in the next TR)? 

Regarding the concurrent event, I understand that I can then code overlapping events in the design matrix (even if their BOLD response might not be additive)?

Thanks again for you help!

George

**Comment by kendrickkay** [comment]
Hi George,

Perhaps a few clarifications to make sure we are on the same page:

- 1-s is a reasonable "short" event duration. The main reason that going faster doesn't really do much is that the shape of the BOLD response to a very brief stimulus (e.g. < 0.1 s) is not very different from that of a 1-s stimulus.  (In the attached screenshot, blue is 0.1s, red is 1.0s.)   Of course, note that this refers only to the shape --- the amplitude of the predicted response to a 1-s stimulus will be much larger.

- For long duration events (e.g. 7.3 s), it depends on what you believe is the associated neural response.  If you believe the neural response is also extended (e.g. 7.3 s), then the idea is that you could model the response as a series of short responses.

- The TR is just the sampling rate that the fMRI data come in at.  It is theoretically distinct from the actual duration of the stimulus/neural events.  Hence, I am not sure exactly what you mean by "splitting"...

- Regarding overlapping events, perhaps I was not being very clear, but my first reaction is that this is a challenge to deal with, and GLMsingle is not suited for that sort of scenario.  It would be more straightforward (with respect to GLMsingle) to just treat cases of "overlapping" events as just instances of a new type of event.

Kendrick





￼
> Thank you so much for your fast reply! I can understand that there will be rounding of events to the closest 1 second so as to account for varying durations (after splitting longer duration to smaller events).
> 
> I just have some follow up questions in regards to that:
> 
> In terms of the duration of the neural events, does it then make sense to make the short events in the order of 1 second because making them shorter would not make the function of the BOLD response reliable anymore? How would I model some events that I have with duration of 7.3 seconds for example?
> Does the TR matter in terms of the matrix of events (i.e. 1.5 TR means 1 second events need to be split in half in the next TR)?
> Regarding the concurrent event, I understand that I can then code overlapping events in the design matrix (even if their BOLD response might not be additive)?
>

**Reply by gtertikas** [reply]
Hi Kendrick,

Thanks a lot for the clarifications. Your point around the concordant button press makes sense. I also understand now that the TR is independent to the stimulus onset. I had misunderstood originally that when you input the design matrix, each event needed to be in reference to the TRs but if it is just in reference to the onset time then this should be fine. Is that indeed the case then? 

May I ask one last thing that I might have also misunderstood; My question essentially was what happens if the duration of the event is not perfectly allinged with the 1s model of the BOLD response. For example if it is shorter (i.e. less than 0.5 sec), would I round this down to a null event? How would I account for this in the design matrix esentially?


Thank you once more for all your help,

George

**Comment by kendrickkay** [comment]
Hi George - 

> independent to the stimulus onset. I had misunderstood originally that when you input the design matrix, each event needed to be in reference to the TRs but if it is just in reference to the onset time then this should be fine. Is that indeed the case then?

The format that GLMsingle requires is a design matrix that is computed at the sampling rate of the TR.  So, when you specify when events occur, this is specified in integer multiples of the TR.  (Sometimes people have cases where their events don't quite align with TRs. This scenario is discussed on the FAQ.)

> May I ask one last thing that I might have also misunderstood; My question essentially was what happens if the duration of the event is not perfectly allinged with the 1s model of the BOLD response. For example if it is shorter (i.e. less than 0.5 sec), would I round this down to a null event? How would I account for this in the design matrix esentially?
> 
Within some range, short events all give rise to very similar predicted BOLD timecourses.  So, if you have a short 0.5 s event, you could just treat this as a 1-s event, and this should work perfectly fine as far as GLMsingle is concerned.


Kendrick
